### PR TITLE
Renovate/major sentry javascript monorepo

### PR DESCRIPTION
**Pull request by trixobird**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

_Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of
the pull request._

## Description

This PR makes the sentry plugin compatible with Sentry v8 SDK.

Fixes # (2193)

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)

## How Has This Been Tested?

Have not written any new tests, have not modified any existing tests.

- [x] `sentry.spec.ts`

**Test Environment**:

- OS: MacOS 14.5 Sonoma
- `@envelop/sentry`:
- NodeJS:

## Checklist:

- [x] I have followed the
      [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the
      style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Resources: 
- https://docs.sentry.io/platforms/javascript/migration/v7-to-v8/v8-new-performance-api/
- https://docs.sentry.io/platforms/javascript/tracing/instrumentation/custom-instrumentation/

### CODE REVIEW

**Comment by gajus** [comment]
Wonderful!

**Comment by gajus** [comment]
@dotansimha Any chance of getting this across the line?

**Comment by gajus** [comment]
@n1ru4l Anything missing to merge this?

**Comment by Boeing787** [comment]
Hey folks @n1ru4l @dotansimha -- would love to see @trixobird 's work get merged here! Looks like it is passing all tests and would unblock us from adopting V8 tracing 🙏 Let me know if you need any support with that 👍

**Comment by FezVrasta** [comment]
Really looking forward to this. Could you guys prioritize this PR please?

edit: for now I published this as a fork on `@fezvrasta/envelop-sentry`.

File: packages/plugins/immediate-introspection/package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
   "typings": "dist/typings/index.d.ts",
   "peerDependencies": {
     "@envelop/core": "^5.0.0",
-    "@sentry/node": "^6 || ^7",
+    "@sentry/node": "^8.0.0",
```
Comment (enisdenjo) [inline packages/plugins/immediate-introspection/package.json:51]: Wouldn't this change be considered breaking as v6 and v7 is not supported anymore?

File: .changeset/@envelop_sentry-2235-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@envelop/sentry': patch
```
Comment (klippx) [inline .changeset/@envelop_sentry-2235-dependencies.md]:
```suggestion
'@envelop/sentry': major
```

**Review by enisdenjo — CHANGES_REQUESTED** [review]
This looks good to me (after applying [this suggestion](https://github.com/n1ru4l/envelop/pull/2277#discussion_r1784355284)). Please rebase so that we can go ahead and merge this.

**Comment by trixobird** [comment]
@enisdenjo rebased and changed the changeset. Let me know if there is anythig else that I can do at this point.

File: .changeset/@envelop_immediate-introspection-2235-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+'@envelop/immediate-introspection': patch
```
Comment (enisdenjo) [inline .changeset/@envelop_immediate-introspection-2235-dependencies.md]:
```suggestion
'@envelop/immediate-introspection': major
```

**Review by enisdenjo — CHANGES_REQUESTED** [review]

File: .changeset/@envelop_immediate-introspection-2235-dependencies.md (hunk 3 above)
Comment (trixobird) [reply .changeset/@envelop_immediate-introspection-2235-dependencies.md]: done!

Comment (trixobird) [reply .changeset/@envelop_immediate-introspection-2235-dependencies.md]: hi @enisdenjo, let me know if there is anythig else that I can do at this point.

**Review by enisdenjo — APPROVED** [review]

**Pull request merged**